src/script/conversation/ConversationRepository.ts

```diff
@@ -467,6 +468,35 @@ export class ConversationRepository {
     await this.userRepository.refreshUsers(unavailableUsers.map(user => user.qualifiedId));
   }
 
+  public async refreshAllConversationsUnavailableParticipants(): Promise<void> {
+    const allUnavailableUsers = this.conversationState
+      .conversations()
+      .flatMap(conversation => conversation.allUserEntities().filter(user => !user.isAvailable()));
+
+    if (!allUnavailableUsers.length) {
+      return;
+    }
+    await this.userRepository.refreshUsers(allUnavailableUsers.map(user => user.qualifiedId));
+  }
+
+  /**
+   * Refresh missing conversations and unavailable users metadata every 3 hours
+   * @Note Federation only
+   */
+  private readonly scheduleMissingUsersAndConversationsMetadataRefresh = () => {
+    if (!this.core.backendFeatures.isFederated) {
+      return;
+    }
```

Comment:
if we do nothing in this case, maybe let's just not call this function at all, so wrap it with the condition:
```ts
if(this.core.backendFeatures.isFederated){
  scheduleMissingUsersAndConversationsMetadataRefresh()
}
```
wdyt?

Reply:
https://github.com/wireapp/wire-webapp/pull/15158/commits/6c6514c894005c97d7b11485a64f5759183e668c